### ISSUE TITLE
Sluggish mobile load performance

### ISSUE THREAD

**Issue by ksambhi** [davwheat/world-cup-of-tocs#30]
Due to the number of graphs being loaded and displayed at once, on mobile performance can be slow when loading the website. We should probably fix this so the site loads faster. I can think of a few solutions:

- SSR? (no idea if it would help)
- Render the graphs serially, rather than in parallel. E.g. load and render serially in this order:
   - The current year
      - The current matches
      - The previous round played
      - The round before that
      - etc.
   - The previous year
      - The final
      - The semi final
      - The quarter final
      - The R16
      - The R32
   - The year before that (using the same order above)
- Render graphs when they are scrolled to (is this called lazy-loading?)

**Comment by davwheat** [comment]
> Render graphs when they are scrolled to (is this called lazy-loading?)

Yes, and this would be my preferred solution. I've used it in another project in the past: https://github.com/davwheat/personal-portfolio/blob/main/src/components/Canvas.tsx

The main thing is the useVisible hook, though.
